components/tenant-fetcher/internal/tenant/converter.go

```diff
@@ -14,21 +16,33 @@ func NewConverter() *converter {
 func (c *converter) ToEntity(in model.TenantModel) tenant.Entity {
 	return tenant.Entity{
 		ID:             in.ID,
-		Name:           in.TenantId,
+		Name:           in.Name,
 		ExternalTenant: in.TenantId,
-		ProviderName:   in.TenantProvider,
+		Parent:         NewNullString(in.ParentInternal),
+		Type:           in.Type,
+		ProviderName:   in.Provider,
 		Status:         in.Status,
 	}
 }
 
-func (c *converter) FromEntity(in *tenant.Entity) *model.TenantModel {
-	if in == nil {
-		return nil
-	}
-	return &model.TenantModel{
+func (c *converter) FromEntity(in tenant.Entity) model.TenantModel {
+	return model.TenantModel{
 		ID:             in.ID,
+		Name:           in.Name,
 		TenantId:       in.ExternalTenant,
-		TenantProvider: in.ProviderName,
+		ParentInternal: in.Parent.String,
+		Type:           in.Type,
```

Comment:
Ant the same for the Type (as it may be nil now)

Reply:
I'm not sure whether Type should be a nullable field. How does it sound if we add an additional Type type `empty`/`none` to indicate an empty tenant, so that we enforce this column to be always non-null? For this to work, it would be also necessary to modify the enum in the SQL migration.